tools/update-blog-dates.js

```diff
@@ -40,23 +40,19 @@ async function getAllBlogFiles() {
 }
 
 /**
- * Gets the list of staged blog post files from git.
+ * Gets the list of staged blog post files from command line arguments.
  * @returns {Promise<string[]>} Array of absolute paths to staged .md files in the blog directory.
  */
 async function getStagedBlogFiles() {
-	const stagedFiles = execSync(
-		"git diff --staged --name-only --diff-filter=ACMRT",
-	)
-		.toString()
-		.trim()
-		.split("\n");
-
-	return stagedFiles
-		.filter(
-			file =>
-				file.startsWith("src/content/blog/") && file.endsWith(".md"),
-		)
-		.map(file => path.resolve(process.cwd(), file));
+	const stagedFiles = process.argv.slice(2);
+
+	return stagedFiles.filter(file => {
+		const relativePath = path.relative(process.cwd(), file);
+		return (
+			relativePath.startsWith(path.normalize("src/content/blog/")) &&
+			relativePath.endsWith(".md")
+		);
+	});
```

Comment:
It's safer to use `path` methods to determine if a file is a blog post. Something like this:

```suggestion
    const blogDir = path.resolve("src/content/blog");

	return stagedFiles.filter(file => {
		const dirPath = path.dirname(file);
		
		return (
			dirPath.startsWith(blogPath)) &&
			file.endsWith(".md")
		);
	});
```